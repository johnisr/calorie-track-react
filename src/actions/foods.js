import uuid from 'uuid';
import database from '../firebase/firebase';
import moment from 'moment';

export const addFood = food => ({
  type: 'ADD_FOOD',
  food: { id: uuid(), ...food },
});

export const startAddFood = (data = {}) => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    const {
      name = '',
      amount = '0',
      unit = 'grams',
      carbohydrates = 0,
      protein = 0,
      fat = 0,
      calories = 0,
      timesUsed = 0,
      createdAt = moment(),
    } = data;
    const food = { name, amount, unit, carbohydrates, protein, fat, calories, timesUsed, createdAt };
    try {
      const res = await database.ref(`users/${uid}/foods`).push(food);
      dispatch(addFood({ id: res.key, ...food }));
    } catch (e) {

    }
  };
}

export const editFood = (id, updates) => ({
  type: 'EDIT_FOOD',
  id,
  updates,
});

export const startEditFood = (id, updates) => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    try {
      await database.ref(`users/${uid}/foods/${id}`).update(updates);
      dispatch(editFood(id, updates));
    } catch (e) {
      
    }
  };
};

export const removeFood = ({ id }) => ({
  type: 'REMOVE_FOOD',
  id,
});

export const startRemoveFood = ({ id }) => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    try {
      await database.ref(`users/${uid}/foods/${id}`).remove();
      dispatch(removeFood({ id }));
    } catch (e) {
      
    }
  };
};

export const setFoods = foods => ({
  type: 'SET_FOODS',
  foods,
});

export const startSetFoods = () => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;

    try {
      const snapshot = await database.ref(`users/${uid}/foods`).once('value');
      const foods = [];
      snapshot.forEach((food) => {
        foods.push({
          id: food.key,
          ...food.val(),
        });
      });
      dispatch(setFoods(foods));
    } catch (e) {

    }
  };
};