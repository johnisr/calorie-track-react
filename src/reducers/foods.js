const foodsDefaultState = [];

export default (state = foodsDefaultState, action) => {
  switch (action.type) {
    case 'ADD_FOOD':
      return [...state, action.food];
    case 'EDIT_FOOD':
      return state.map((food) => {
        if (food.id === action.id) {
          return {
            ...food,
            ...action.updates,
          };
        }
        return food;
      });
    case 'REMOVE_FOOD':
      return state.filter(({ id }) => id !== action.id);
    case 'SET_FOODS':
      return action.foods;
    default:
      return state;
  }
}