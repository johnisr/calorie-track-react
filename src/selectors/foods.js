export default (foods, { name, sortBy, maxFoodsShown, offset }) => {
  return foods
    .filter((food) => {
      const nameMatch = food.name.toLowerCase().includes(name.toLowerCase());
      return nameMatch;
    })
    .sort((a, b) => {
      if (sortBy === 'usage') {
        return a.timesUsed < b.timesUsed ? 1 : -1;
      } else if (sortBy === 'date') {
        return a.createdAt < b.createdAt ? 1 : -1;
      }
    })
    .slice((offset * maxFoodsShown), (offset * maxFoodsShown) + maxFoodsShown);
}