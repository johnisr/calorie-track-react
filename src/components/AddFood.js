import React from 'react';
import { connect } from 'react-redux';
import FoodForm from './FoodForm';
import { startAddFood } from '../actions/foods';

export class AddFood extends React.Component {
  handleSubmit = (food) => {
    this.props.startAddFood({
      ...food,
    });
  }
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Food</h1>
          </div>
        </div>
        <div className="content-container">
          <FoodForm
            handleSubmit={this.handleSubmit}
          />
        </div>
      </div>
    )
  }
}
const mapDispatchToProps = dispatch => ({
  startAddFood: food => dispatch(startAddFood(food)),
});

export default connect(undefined, mapDispatchToProps)(AddFood);
