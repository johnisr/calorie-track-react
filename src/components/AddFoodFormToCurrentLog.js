import React from 'react';
import { connect } from 'react-redux';
import FoodForm from './FoodForm';
import { addFoodToCurrentLog } from '../actions/currentLog';

export class AddFoodFormToCurrentLog extends React.Component {
  handleSubmit = (food) => {
    this.props.addFoodToCurrentLog(food);
  }
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Food To Log</h1>
          </div>
        </div>
        <div className="content-container">
          <FoodForm
            handleSubmit={this.handleSubmit}
          />
        </div>
      </div>
    )
  }
}
const mapDispatchToProps = dispatch => ({
  addFoodToCurrentLog: food => dispatch(addFoodToCurrentLog(food)),
});

export default connect(undefined, mapDispatchToProps)(AddFoodFormToCurrentLog);
