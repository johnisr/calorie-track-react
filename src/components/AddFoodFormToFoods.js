import React from 'react';
import { connect } from 'react-redux';
import FoodForm from './FoodForm';
import { startAddFood } from '../actions/foods';

export class AddFoodFormToFoods extends React.Component {
  handleSubmit = (food) => {
    console.log('Submitting to Foods');
    console.log(food);
    this.props.startAddFood(food);
  }
  render() {
    return (
      <div className="pane">
        <h1 className="pane__title">Add Food</h1>
        <FoodForm handleSubmit={this.handleSubmit} />
      </div>
    )
  }
}
const mapDispatchToProps = dispatch => ({
  startAddFood: food => dispatch(startAddFood(food)),
});

export default connect(undefined, mapDispatchToProps)(AddFoodFormToFoods);
