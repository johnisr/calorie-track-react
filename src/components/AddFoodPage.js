import React from 'react';
import { connect } from 'react-redux';
import FoodForm from './FoodForm';
import { addFood } from '../actions/foods';

export class AddFoodPage extends React.Component {
  handleSubmit = (food) => {
    this.props.addFood({
      ...food,
    });
    this.props.history.push('/home');
  }
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Food</h1>
          </div>
        </div>
        <div className="content-container">
          <FoodForm
            handleSubmit={this.handleSubmit}
          />
        </div>
      </div>
    )
  }
}
const mapDispatchToProps = dispatch => ({
  addFood: food => dispatch(addFood(food)),
});

export default connect(undefined, mapDispatchToProps)(AddFoodPage);
