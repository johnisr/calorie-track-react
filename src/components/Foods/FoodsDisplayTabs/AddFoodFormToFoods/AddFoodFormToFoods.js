import React from 'react';
import { connect } from 'react-redux';
import FoodForm from '../../../FoodForm/FoodForm';
import { startAddFood } from '../../../../actions/foods';

export class AddFoodFormToFoods extends React.Component {
  handleSubmit = (food) => {
    this.props.startAddFood(food);
  }
  render() {
    return (
      <FoodForm handleSubmit={this.handleSubmit} title={'Add Food'} />
    )
  }
}
const mapDispatchToProps = dispatch => ({
  startAddFood: food => dispatch(startAddFood(food)),
});

export default connect(undefined, mapDispatchToProps)(AddFoodFormToFoods);
