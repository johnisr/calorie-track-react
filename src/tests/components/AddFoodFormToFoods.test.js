import React from 'react';
import { shallow } from 'enzyme';
import { AddFoodFormToFoods } from '../../components/AddFoodFormToFoods';
import foods from '../fixtures/foods';

let startAddFood, wrapper;
beforeEach(() => {
  startAddFood = jest.fn();
  wrapper = shallow(<AddFoodFormToFoods startAddFood={startAddFood}/>);
});

it('should render AddFood correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should handle onSubmit', () => {
  wrapper.find('FoodForm').prop('handleSubmit')(foods[1]);
  expect(startAddFood).toHaveBeenLastCalledWith(foods[1]);
});