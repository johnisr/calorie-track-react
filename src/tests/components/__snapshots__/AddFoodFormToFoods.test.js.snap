// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddFood correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddFoodFormToFoods
    startAddFood={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="pane__title"
>
          Add Food
</h1>,
        <FoodForm
          handleSubmit={[Function]}
/>,
      ],
      "className": "pane",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Food",
          "className": "pane__title",
        },
        "ref": null,
        "rendered": "Add Food",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="pane__title"
>
            Add Food
</h1>,
          <FoodForm
            handleSubmit={[Function]}
/>,
        ],
        "className": "pane",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Food",
            "className": "pane__title",
          },
          "ref": null,
          "rendered": "Add Food",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
